client/state/data-layer/wpcom/concierge/schedules/shifts/test/from-api.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
```

Comment:
> These tests depend a lot on the data-layer internals. 

What do you mean? Shouldn't we be testing our `fromApi`s

Reply:
> What do you mean? Shouldn't we be testing our fromApis

Sorry for the initial, rather vague comment. I have some opinions here, but they're not fully baked yet and I'm not much of a testing expert, so I ended up not expressing much 🙁 

> transformShift doesn't actually get used by the data-layer.  In a sense, `makeParser` is just a helper for making a `fromApi`

💯  - I said `data-layer` and was really referring to `makeParser`.

* `fromApi()` should probably be called in the tests, I'm not trying to discourage that. However, it seems that what we're really doing is testing `transformShift` after it comes out the other side of `makeParser`. There's value in knowing that `fromApi` doesn't fail or warn about a schema parsing error when called (this happened recently and went undetected for a while). `fromApi` might be a good candidate for a snapshot test (p4TIVU-8Hv-p2)
* `transform` and `transformShift` are the only functions entirely under our control
* `transform` is just mapping our `transformShift`, there's not much value testing that.
* `transformShift` is the only thing we've really implemented here of any interest. Granted, it's very simple, but I'd argue it's what we should be targeting with tests.
* In my first (and only) experience implementing `fromApi` like this, I also added tests that essentially targeted the schema. I ended up removing those because they seem to add little value.

I'm happy to dialog more on this and try to get these opinions fully baked, they're not strongly help 🙂

Reply:
That's an insightful discussion above, @sirreal @samouri 🙇 

In the end, I chose to add a simple test for `transformShift()` in 5644f97, since I feel it makes sense to test this individual transforming function. For example, it actually picks only the expected fields, and I think that's not something covered by `makeParser()`. 

Hopefully that makes sense to you two :)

Reply:
👍  Works for me @southp 🙂